test/unit/transport/GrpcTransport.spec.js

```
@@ -0,0 +1,342 @@
+    dapiAddress = new DAPIAddress(host);
+
+    addressProviderMock = {
+      getLiveAddress: this.sinon.stub().resolves(dapiAddress),
+      hasLiveAddresses: this.sinon.stub().resolves(false),
+    };
+
+    globalOptions = {
+      retries: 0,
+    };
+
+    createAddressProviderFromOptionsMock = this.sinon.stub().returns(null);
+
+    grpcTransport = new GrpcTransport(
+      createAddressProviderFromOptionsMock,
+      addressProviderMock,
+      globalOptions,
+    );
+
+    // noinspection JSUnresolvedFunction
+    url = grpcTransport.makeGrpcUrlFromAddress(dapiAddress);
+  });
+
+  describe('#request', () => {
+    let method;
+    let clientClassMock;
+    let requestMessage;
+    let options;
+    let data;
+    let requestFunc;
+
+    beforeEach(function beforeEach() {
+      data = 'result';
+      method = 'method';
+      requestMessage = 'requestMessage';
+      options = {
+        option: 'value',
+      };
+
+      requestFunc = this.sinon.stub().resolves(data);
+
+      clientClassMock = this.sinon.stub().returns({
+        [method]: requestFunc,
+      });
+
+      addressProviderMock.hasLiveAddresses.resolves(true);
+
+      globalOptions = {
+        retries: 1,
+      };
+
+      grpcTransport = new GrpcTransport(
+        createAddressProviderFromOptionsMock,
+        addressProviderMock,
+        globalOptions,
+      );
+    });
+
+    it('should make a request', async () => {
+      const receivedData = await grpcTransport.request(
+        clientClassMock,
+        method,
+        requestMessage,
+        options,
+      );
+
+      expect(receivedData).to.equal(data);
+      expect(createAddressProviderFromOptionsMock).to.be.calledOnceWithExactly(options);
+      expect(clientClassMock).to.be.calledOnceWithExactly(url);
+      expect(requestFunc).to.be.calledOnceWithExactly(requestMessage);
+      expect(grpcTransport.lastUsedAddress).to.deep.equal(dapiAddress);
+    });
+
+    it('should throw unknown error', async () => {
```

should throw an unknown error if it happened during the request